lib/std/crypto/gimli.zig

```diff
@@ -40,8 +40,8 @@ pub const State = struct {
 
     pub fn permute(self: *Self) void {
         const state = &self.data;
-        var round = @as(u32, 24);
-        while (round > 0) : (round -= 1) {
+        comptime var round = @as(u32, 24);
+        inline while (round > 0) : (round -= 1) {
```

Comment:
doesn't this blow up code size significantly?
I liked gimli because of how small (in flash) it can be.

Reply:
This adds 9KB on x86_64, 14KB on aarch64.

But since we use Gimli for our CSPRNG, it may be well worth it. When generating random nonces for encryption, our slow PRNG is currently a bottleneck :(

The permutation can easily be reimplemented using generic vector instructions, which should help a little bit both to improve performance and reduce the size.

Reply:
> The permutation can easily be reimplemented using generic vector instructions, which should help a little bit both to improve performance and reduce the size.

Platforms where the code size matters are often without a vector unit, think of the ARM Cortex-m line or some simple RISC-V cores.

Would some partial unrolling (eg. 2x) help reach a compromise here?

Reply:
Maybe you could branch on `if (std.builtin.mode == .ReleaseSmall)`?

Reply:
Yep, branching on `std.builtin.mode` is a good idea.